torch_geometric/nn/models/captum.py

```diff
@@ -79,39 +96,55 @@ def __init__(self, model: torch.nn.Module, mask_type: str, output_id: int,
-        self, *args
-    ) -> Tuple[Dict[NodeType, Tensor], Dict[EdgeType, Tensor], Optional[Dict[
-            EdgeType, Tensor]]]:
+    def _captum_data_to_hetero_data(self, *args) -> Batch:
         """Converts tuple of tensors to `x_dict`, `edge_index_dict` and
-        `edge_mask_dict`."""
+        `edge_mask_dict` and merges them to a batch object."""
 
         if self.mask_type == 'node':
             node_tensors = args[:self.num_node_types]
-            node_tensors = [mask.squeeze(0) for mask in node_tensors]
-            x_dict = dict(zip(self.node_types, node_tensors))
             edge_index_dict = args[self.num_node_types]
         elif self.mask_type == 'edge':
             edge_mask_tensors = args[:self.num_edge_types]
             x_dict = args[self.num_edge_types]
             edge_index_dict = args[self.num_edge_types + 1]
         else:
             node_tensors = args[:self.num_node_types]
-            node_tensors = [mask.squeeze(0) for mask in node_tensors]
-            x_dict = dict(zip(self.node_types, node_tensors))
             edge_mask_tensors = args[self.num_node_types:self.num_node_types +
                                      self.num_edge_types]
             edge_index_dict = args[self.num_node_types + self.num_edge_types]
 
-        if 'edge' in self.mask_type:
-            edge_mask_tensors = [mask.squeeze(0) for mask in edge_mask_tensors]
-            edge_mask_dict = dict(zip(self.edge_types, edge_mask_tensors))
-        else:
-            edge_mask_dict = None
-        return x_dict, edge_index_dict, edge_mask_dict
+        # Merge disjoint graphs into a single graph:
+        data_list = []
+        for i in range(args[0].shape[0]):
+            if self.mask_type != 'edge':
+                node_features = [mask[i].squeeze(0) for mask in node_tensors]
+                x_dict = dict(zip(self.node_types, node_features))
+            if 'edge' in self.mask_type:
+                edge_masks = [mask[i].squeeze(0) for mask in edge_mask_tensors]
+                edge_masks_dict = dict(zip(self.edge_types, edge_masks))
+            else:
+                edge_masks_dict = None
+            # Create HeteroData object:
+            data = HeteroData()
+            for key, value in edge_index_dict.items():
+                data[key].edge_index = value
+            for key, value in x_dict.items():
+                data[key].x = value
+            if edge_masks_dict is not None:
+                for key, value in edge_masks_dict.items():
+                    data[key].edge_mask = value
+            data_list.append(data)
+        return Batch.from_data_list(data_list)
 
     def forward(self, *args):
         # Validate args:
+        batch_size = args[0].shape[0]
+        if type(self.output_idx) == list:
+            msg = "`internal_batch_size` needs to be a multiple of the " \
+                  "number of objects to be evaluated."
+            assert args[0].shape[0] % (len(self.output_idx)) == 0, msg
```

Comment:
```suggestion
            assert batch_size == len(self.output_idx), msg
```
Wouldn't this be true?

Reply:
No, because it can be more than the number of `output_idx`. E.g. you can set the batch_size to be 50 even if you want to explain only 2 node predictions. This can happen if the calculation of the explanation requires several runs trough the model.